sdk/scheduler/src/test/java/com/mesosphere/sdk/framework/ImplicitReconcilerTest.java

```diff
@@ -0,0 +1,50 @@
+package com.mesosphere.sdk.framework;
+
+import java.util.Collections;
+import org.apache.mesos.SchedulerDriver;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.mesosphere.sdk.scheduler.SchedulerConfig;
+
+import static org.mockito.Mockito.*;
+
+public class ImplicitReconcilerTest {
+
+    @Mock private SchedulerConfig mockSchedulerConfig;
+    @Mock private SchedulerDriver mockSchedulerDriver;
+
+    private ImplicitReconciler reconciler;
+
+    @Before
+    public void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+        Driver.setDriver(mockSchedulerDriver);
+
+        reconciler = new ImplicitReconciler(mockSchedulerConfig);
+    }
+
+    @Test
+    public void testSingleThread() {
+        reconciler.disableThreading().start();
+        verify(mockSchedulerDriver).reconcileTasks(Collections.emptyList());
+        try {
+            reconciler.start();
+            Assert.fail("Expected exception from double start");
+        } catch (IllegalStateException e) {
+            // expected
```

Comment:
There's an argument you can pass to the `@Test` annotation for this.  It's what's used in other tests in the SDK for this pattern.

Reply:
e.g.
```
 @Test(expected = IllegalArgumentException.class)
```

Reply:
Switched to `expected` (will be in the following PR). Also broke this out into two test cases: One for the double start (expect error) and another for the single threaded behavior (don't expect error)